src/neighborsearch/kball_searcher.jl

```diff
@@ -0,0 +1,52 @@
+# ------------------------------------------------------------------
+# Licensed under the ISC License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+KBallSearcher(object, neighborhood)
+
+A method for searching neighbors in spatial `object` inside `neighborhood`.
+Used when doing estimation combining k-nearest and radius.
+It searches first for the k-nearest by using the internal kdtree and then
+limiting the results within the `BallNeighborhood` by calling `isneighbor`
+"""
+struct KBallSearcher{S<:AbstractNeighborSearcher} <: AbstractBoundedNeighborSearcher
+  searcher::S
+  nmax::Int
```

Comment:
Maybe the second field of the KBallSearcher is the ball? So you first search with the inner searcher, and then post-process the results with the ball, makes sense? No need to specify `nmax` because that is already inside the inner nearest-neighbor searcher.

Reply:
That is one option. I just tried to reuse as much as possible what is done. For user perspective, it could be easier to just indicate something like this:
solver = Kriging(
    :cu => (variogram=vmodel,maxneighbors=25,radius=50.0)
)
but I guess we don't want to change the API.

Reply:
Yes, I think it is more general to leave the option as neighborhood because then we have other kinds of neighborhood that could be useful that are not norm-balls.